### DISCUSSION TITLE
Suggest the commit range under point when user calls diff commands

### DISCUSSION THREAD

**Discussion by andreyorst** [magit/magit#5000] [Ideas]
I have a small snippet in my configuration that does this and I find it pretty handy when I view the process log of a `git pull`:

```emacs-lisp
  (define-advice magit-list-refnames (:filter-return (refs) range-at-point)
    (require 'thingatpt)
    (if-let ((range (save-match-data
                      (and (thing-at-point-looking-at "[a-f0-9]+\.\.[a-f0-9]+")
                           (match-string 0)))))
        (cons range refs)
      refs))
```

It's a basic advice that inspects the thing at point and if it matches the regular expression it's added to the list of refs.

For example, here's the point stands on `123060841d9..3dac48d1b73` and I call <kbd>d</kbd><kbd>d</kbd> (diff dwim):

![image](https://github.com/magit/magit/assets/19470159/884706c6-d178-423a-bda5-6d0056613506)

The completion then shows me this range and I don't need to copy it, just put the point on it. Feels like a natural thing for DWIM to do.

**Comment by tarsius** [comment]
Magit does that in `magit-revision-mode` (and maybe other places, I don't remember).

Doing it here would be another "I though if it worked like this and, behold, that's what magit does!" feature, but it would also be one of those features that most users would seldom, if ever, notice, and for them it would just be one more feature that makes magit ever so slightly slower.

This particular feature would (probably) not move the needle much on its own, but it is just one of many little things we could add that would make everything a bit slower, so things add up. I am afraid this one doesn't make the cut, not least because I have already decided to not try to make any of the output in the process log prettier and/or more actionable. (E.g., the diff stat also isn't colorized, and there isn't even any support for color output from git hooks.)

**Reply by tarsius** [reply]
Ah wait, we don't have to fontify the buffer, we can just do what you did. Sorry, I am in a bit of a "get of my todo list" kinda mood. Makes sense, will probably do something like that, but it might take a while, since I am stressed out.

**Reply by andreyorst** [reply]
No problem! 

The advice works well enough for me, so if there's no interest in this feature by others I don't mind continuing to use it as is. Or if the above implementation is OK by you, I can send a patch, adding this as a private function, and calling it during the normal `magit-list-refnames` execution.

**Reply by tarsius** [reply]
By the way this already works for revisions, but not ranges.  If there is a revision at point and `magit-diff-dwim` (or a similar command) is reading input in the minibuffer, you can use `next-history-element` to access that.

It doesn't work for revisions yet though. It also doesn't work for commits, appearing in a range, but that's a bug. I'll fix that and implement the same thing for ranges.

**Reply by tarsius** [reply]
Implemented in a3b7cdc486333fad0813f836ad23c7c1d2426328 and 27099441a048e977f3316ec58dcef104ec6637d1.

Note that you might have to press `M-n` twice (if there is a default, the first `M-n` selects that). Actually because there might be multiple defaults, you might have to press `M-n` even more. See also #4740.

**Comment by andreyorst** [comment]
Thanks a lot!
